sql-plugin/src/main/scala/org/apache/spark/sql/rapids/complexTypeCreator.scala

```diff
@@ -27,8 +27,9 @@ import org.apache.spark.sql.internal.SQLConf
 import org.apache.spark.sql.types.{ArrayType, DataType, Metadata, NullType, StringType, StructField, StructType}
 import org.apache.spark.sql.vectorized.ColumnarBatch
 
-case class GpuCreateArray(children: Seq[Expression], useStringTypeWhenEmpty: Boolean)
-    extends GpuExpression {
+case class GpuCreateArray(children: Seq[Expression], useStringTypeWhenEmpty: Boolean,
+    /* use refer type for children whose type can't be determined */
+    referType: Option[DataType] = None) extends GpuExpression {
```

Comment:
Why can it not be determined? each `Expression` has a type associated with it. In what situation is there a `Expression` that says it is the wrong type? Is this because we don't support array literals? The PRs are in progress for array literal support and I am inclined to just hold off on this and all of the work arounds in it until we actually have proper support.

Reply:
Hi @revans2, Sorry for my unclear comment in the code. What I wanted to say is. For the case `array(array(1), array()`, with this PR, it will be converted to
``` console
GpuCreateArray
     |--children(0):  GpuCreateArray
               |-- GpuLiteral(value = 1, dataType=IntegerType).
     |--children(1):  GpuCreateArray no children
```
So the dataType of child(0) will be ArrayType(IntegerType) and the dataType of  child(1) will be `ArrayType(NullType) or ArrayType(StringType) that depends on LEGACY_CREATE_EMPTY_COLLECTION_USING_STRING_TYPE`

Later in somewhere, GpuCreateArray will be resolved and call `checkInputDataTypes` which requires all the children should have the same data type. If false. the GpuCreateArray will become un-resolved and cause some issues. So I add a refer type (from parent element of array) for the case no children and bypass `checkInputDataTypes`.